src/stores/SpaceStore.tsx

```diff
@@ -410,6 +410,22 @@ export class SpaceStoreClass extends AsyncStoreWithClient<IState> {
         });
     }, 100, {trailing: true, leading: true});
 
+    private switchToRelatedSpace = (roomId: string) => {
+        if (this.suggestedRooms.find(r => r.room_id === roomId)) return;
+
+        let parent = this.getCanonicalParent(roomId);
+        if (!parent) {
+            parent = this.rootSpaces.find(s => this.spaceFilteredRooms.get(s.roomId)?.has(roomId));
+        }
+        if (!parent) {
+            const parents = Array.from(this.parentMap.get(roomId) || []);
+            parent = parents.find(p => this.matrixClient.getRoom(p));
+        }
+
+        // don't trigger a context switch when we are switching a space to match the chosen room
```

Comment:
Could you add a comment to `setActiveSpace` or similar which summarises when and when not to use context switch mode? I tried to work it out from reading usages, but I'm still not quite sure... Is `contextSwitch == true` for the case of switching spaces but not the active room...?